File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Define variable "change" as a float
    float change;

    // Get input from user for amount of change owed
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // Define cents
    int centsOwed = round(change * 100);

    // subtract quarters from cent total
    int quartersOwed = floor(centsOwed / 25);
    centsOwed -= quartersOwed * 25;

    // subtract dimes from cent total
    int dimesOwed = floor(centsOwed / 10);
    centsOwed -= dimesOwed * 10;

    // subtract nickels from cent total
    int nickOwed = floor(centsOwed / 5);
    centsOwed -= nickOwed * 5;

    // Add together total of each type of coin
    int coinsTotal = centsOwed + nickOwed + dimesOwed + quartersOwed;

    // Print answer and finish program
    printf("%i\n", coinsTotal);
    return 0;

}

File: pset6/cash/cash.py
import math
from cs50 import get_float

# Ask user for input
change = get_float("Changed owed: ")

# Check if user gives correct input
while change <= 0:
    change = get_float("Changed owed: ")

# Calculate amount of quarters 
quarters = math.floor(change * 4)
change = round(change - quarters * 0.25, 2)

# Calculate amount of dimes
dimes = math.floor(change * 10)
change = round(change - dimes * 0.1, 2)

# Calculate amount of nickels
nickels = math.floor(change * 20)
change = round(change - nickels * 0.05, 2)

# Calculate amount of pennies
pennies = math.floor(change * 100)

# Print total amount of coins 
print(quarters + dimes + nickels + pennies)